src/newTest/java/org/opensearch/test/GenericIntegrationTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.test;
+
+import org.apache.http.HttpStatus;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.opensearch.test.framework.TestIndex;
+import org.opensearch.test.framework.TestSecurityConfig;
+import org.opensearch.test.framework.TestSecurityConfig.Role;
+import org.opensearch.test.framework.cluster.ClusterConfiguration;
+import org.opensearch.test.framework.cluster.LocalCluster;
+import org.opensearch.test.framework.cluster.TestRestClient;
+import org.opensearch.test.framework.cluster.TestRestClient.HttpResponse;
+
+import com.fasterxml.jackson.core.JsonPointer;
+
+/**
+ * WIP
+ * Generic test class that demonstrates how to use the test framework to 
+ * set up a test cluster with users, roles, indices and data, and how to
+ * implement tests. One main goal here is to make tests self-contained.
+ */
+public class GenericIntegrationTest extends AbstractIntegrationTest {
+	    
+    // define indices used in this test
+    private final static TestIndex INDEX_A = TestIndex.name("index-a").build();
+    private final static TestIndex INDEX_B = TestIndex.name("index-b").build();
+    
+    private final static TestSecurityConfig.User INDEX_A_USER = new TestSecurityConfig.User("index_a_user")
+            .roles(new Role("index_a_role").indexPermissions("*").on(INDEX_A).clusterPermissions("*"));
+
+
+    // build our test cluster as a ClassRule
+    @ClassRule
+    public static LocalCluster cluster = new LocalCluster.Builder().clusterConfiguration(ClusterConfiguration.THREE_MASTERS)
+    	.authc(AUTHC_HTTPBASIC_INTERNAL)
+    	.users(USER_ADMIN, INDEX_A_USER)
+    	.indices(INDEX_A, INDEX_B).build();
+
+    @Test
+    public void testAdminUserHasAccessToAllIndices() throws Exception {
+        try (TestRestClient client = cluster.getRestClient(USER_ADMIN)) {
+            HttpResponse response = client.get("*/_search?pretty");
+            Assert.assertEquals(response.getStatusCode(), HttpStatus.SC_OK);
+        }        
+    }
+
+    @Test
+    public void testIndexAUserHasOnlyAccessToIndexA() throws Exception {
+        try (TestRestClient client = cluster.getRestClient(INDEX_A_USER)) {
+            HttpResponse response = client.get("index-a/_search?pretty");
```

Comment:
How about being able to get a search request from a TestIndex object?  that would reduce the amount of string-fu needed to author these tests

Reply:
Nice idea. However, we will need quite few different queries later, so such a functionality would need to be quite flexible. Maybe we can combine it with `QueryBuilder` in some way?

Reply:
If we are creating `TestIndex` it seems like a lost opportunity not to be able to put functionality like this inside of that class.  While it should be possible to directly author urls to the service, I think it would avoid duplication and create appropriate coupling between test cases removing copy-pasta related errors of which there are many in the existing test cases.

#### Possible 
```java
@Test
public void testClusterSettingsUpdate() {
   final ClusterSettings newSettings = ClusterSettings.builder()....build();
   try (TestRestClient client = cluster.getRestClient(LIMITED_USER)) {
      newSettings.putExpectingStatus(client, HttpStatusCode.Forbidden);
   } 

   try (TestRestClient client = cluster.getRestClient(ADMIN_USER)) {
      newSettings.putExpectingStatus(client, HttpStatusCode.NoBody)
   }
}
```

Reply:
I guess in the current nomenclature this would be `TestClusterSettings`, and I'm thinking its an imperfect example because its represents a singleton... that said the pattern is that modeled object (e.g. TestIndex) would know its interaction model and support similar conventions.  in this case the test case knowing the actual url to the object isn't critical, only knowing that the `put` action is done and the expected results are returned.